README.md

```diff
@@ -135,81 +135,108 @@ Push the `local.tfstate.BACK`.
+## Understanding move-after and move-before
+
+Consider root environment (name `1`), represented as list element:
+
+![](docs/list.png)
+
+There are two ways to split it:
 
-Source root:
+1. By putting some of its contents AFTER itself.
+  ![](docs/list-append.png)
+2. By putting some of its contents BEFORE itself.
+  ![](docs/list-prepend.png)
+
+Move AFTER is more frequent and easier to reason about.
+
+On the other hand, you will know when you need to move BEFORE `terraform plan` in the BEFORE module (1' in the figure above) will fail with a message similar to this one:
 
 ```
-$ cd $SRC_ROOT_MODULE
-$ terraform workspace select $WS
-$ terraform plan -no-color 2>&1 | tee src-plan.txt
+Error: Unsupported attribute
+│
+│ on main.tf line 11, in resource "null_resource" "res2":
+│ 11:     dep = data.terraform_remote_state.prepend_p0.outputs.res1_id
+│ data.terraform_remote_state.prepend_p0.outputs is object with 1 attribute "pet"
+│
+│ This object does not have an attribute named "res1_id".
+╵
+```
+
+The error is because we didn't run terraform apply in `p0`, so `p0.outputs` doesn't have yet the attribute `res1_id`.
+
+When this happens and you convince yourself that this is expected and not an error on your part, you can still move the state, by using command `move-before`.
+
+## Collect information and remote state
+
+Perform all operations in `topdir`, the directory containing the two root modules.
+
+Something like:
 
-$ terraform state pull > local.tfstate
-$ cp local.tfstate local.tfstate.BACK
+```
+topdir/
+    BEFORE/    <== root module
+    AFTER/     <== root module
 ```
 
-Destination root:
+```
+$ terraform -chdir=BEFORE state pull > BEFORE.tfstate
+$ terraform -chdir=AFTER  state pull > AFTER.tfstate
+```
 
 ```
-$ cd $DST_ROOT_MODULE
-$ terraform workspace select $WS
-$ terraform plan -no-color 2>&1 | tee dst-plan.txt
+$ terraform -chdir=BEFORE plan -no-color > BEFORE.tfplan   # this for any move
+$ terraform -chdir=AFTER  plan -no-color > AFTER.tfplan    # this ONLY for move-before
+```
 
-$ terraform state pull > local.tfstate
-$ cp local.tfstate local.tfstate.BACK
+```
+$ cp BEFORE.tfstate BEFORE.tfstate.BACK
+$ cp AFTER.tfstate AFTER.tfstate.BACK
 ```
 
 The backups are needed to recover in case of errors. They must be done now.
 
 ## Generate migration scripts
 
-Take as input the two Terraform plans `src-plan.txt`, `dst-plan.txt`, the two local state files in the corresponding directories and generate UP and DOWN migration scripts.
+Take as input the one or two Terraform plans (BEFORE.tfplan and AFTER.tfplan) and the two state files (BEFORE.tfstate and AFTER.tfstate) and generate the 01-migrate-foo_up.sh and -1-migrate-foo_down.sh migration scripts.
 
-Assuming the following directory layout, where `repo` is the top-level directory and `src` and `dst` are the two Terraform root modules:
+If move-after:
 
 ```
-repo/
-├── src/
-├── dst/
+$ terravalet move-after  --script=01-migrate-foo --before=BEFORE --after=AFTER
```

Comment:
Should it be: `$ terravalet move-after  --script=01-migrate-foo --before=BEFORE/BEFORE.tfstate--after=AFTER/AFTER.tfstate`?

Reply:
Actually no. Everything should be done from top level (its explained above.)
But `terravalet move-after --script=01-migrate-foo --before=BEFORE.tfplan --after=AFTER.tfplan` would be more correct?

Reply:
Again, no. It was clear from other PR. Sorry for the noise.

Reply:
No problems!